MOxUnit/moxunit_util_elem2str.m

```diff
@@ -18,20 +18,54 @@
 %     to capture the output from disp in a variable; hence only the class
 %     of the first input is shown
 %
-% NNO Jan 2014
+% NNO, SCL 2014-2015
 
     if nargin<2
         max_chars=100;
     end
 
-    if moxunit_util_platform_is_octave()
-        elem_str=sprintf('<%s>\n',class(elem));
-    else
-        % Octave does not support evalc
+    % Strings are trivally supported by leaving them as-is
+    if ischar(elem)
+        % Encapsulate in quotation marks
+        elem_str=['''' elem ''''];
+
+    elseif isnumeric(elem)
+        % If the element is numeric, we can use mat2str
+        elem_str=mat2str(elem);
+
+    elseif exist('evalc', 'builtin')
+        % If we have `evalc`, we can just trust the `display` function
+        % provided by the environment to format the string correctly
+        % and capture it with `evalc`.
         elem_str=evalc('disp(elem)');
 
         % remove trailing newlines
         elem_str=elem_str(1:(end-2));
+
+        if iscell(elem)
+            % Encapsulate output in curly braces if it is a cell.
+            % These are not included by disp.
+            elem_str = ['{ ' strtrim(elem_str) ' }'];
+        end
+
+    elseif iscell(elem)
+        % Handle cells, without using `evalc`
+        % Just show the size of the cell and indicate the type with
+        % curly braces. This is how MATLAB displays cells in cells.
+        kk = size(elem);
+        xx = repmat('x', 1, 2*length(kk)-1);
+        xx(1:2:end) = arrayfun(@num2str, kk);
+        elem_str = ['{' xx ' cell}'];
+
+    else
+        % Octave does not support evalc, so we need a work around.
+        % As a fall back, we just show the class of the element
+        elem_str=sprintf('*%s instance*\n',class(elem));
```

Comment:
I would prefer to keep this function simple and stupid, and just use class in all cases. It would be fine with me to remove the '<' and '>' characters. Maybe even try to determine the size (within a try-catch statement)